bluesky/suspenders.py

```diff
@@ -525,3 +525,123 @@ def _get_justification(self):
                 )
         return ': '.join(s for s in (just, self._tripped_message)
                          if s)
+
+
+class SuspendWhenChanged(SuspenderBase):
+    """
+    Suspend when the monitored value deviates from the expected.
+    
+    Only resume if allowed AND when monitored equals expected.
+    
+    .. note: this suspender is designed to require bluesky restart if value changes
+    
+    USE CASE:
+    
+    :class:`~SuspendWhenChanged()` is useful when ``signal`` is an EPICS enumeration 
+    ([``mbbo`` record](https://wiki-ext.aps.anl.gov/epics/index.php/RRM_3-14_Multi-Bit_Binary_Output))
+    used with a multi-instrument facility.  
+    Choices are the instruments allowed to control any shared hardware.
+    The ``signal``, set by instrument staff outside of bluesky,
+    names which instrument is allowed to control the hardware.
+    Other instruments not matching ``signal`` are expected **not** to 
+    control the hardware (they could use simulators instead or not operate
+    the shared hardware).
+    
+    Since a decision of hardware *vs.* simulators is made at the 
+    time a bluesky session starts and ophyd objects are first created, the 
+    session needs to be aware immediately if the ``signal`` is changed.  
+    The default value of ``allow_resume=False`` defends this decision.
+    If there is a mechanism engineered to toggle ophyd signals between 
+    hardware and simulators, one might consider ``allow_resume=False``.
+    
+    USAGE::
+
+        # pause if this value changes in our session
+        # note: this suspender is designed to require Bluesky restart if value changes
+        suspend_instrument_in_use = SuspendWhenChanged(instrument_in_use)
+        RE.install_suspender(suspend_instrument_in_use)
+    
+    Parameters
+    ----------
+
+    signal : `ophyd.Signal`
+        The signal to watch for changes to determine if the
+        scan should be suspended
+
+    expected_value : number (float or int)
+        RunEngine operations will be suspended when signal deviates
+        from this value.  If `None` (default), set to value of 
+        ``signal`` when object is created.
+
+    allow_resume : bool
+        Should RunEngine be allowed to resume once ``signal.value == expected``
+        again?  Default value of ``False`` is expected for intended use case.
+    
+    sleep : float, optional
+        How long to wait in seconds after the resume condition is met
+        before marking the event as done.  Defaults to 0
+
+    pre_plan : iterable or iterator or generator function, optional
+            a generator, list, or similar containing `Msg` objects
+
+    post_plan : iterable or iterator or generator function, optional
+            a generator, list, or similar containing `Msg` objects
+
+    tripped_message : str, optional
+        Message to include in the trip notification
+
```

Comment:
```suggestion
Notes
------
```

This should get numpydoc to not treat this as a parameter.

Reply:
Shouldn't it have the same number of dashes as the number of letters (5)? The number of dashes is 6...

Reply:
reST allows the underline to extend beyond.  What about numpydoc?

Reply:
I am used to working with monopsace fonts....

Sphinx and numpydoc vary in strictness based on version....